kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/partition/FixedLongRangeFieldPartitioner.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright 2013 Cloudera Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kitesdk.data.spi.partition;
+
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Predicate;
+import com.google.common.collect.DiscreteDomains;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+import org.kitesdk.data.spi.FieldPartitioner;
+import org.kitesdk.data.spi.predicates.Exists;
+import org.kitesdk.data.spi.predicates.In;
+import org.kitesdk.data.spi.predicates.Predicates;
+import org.kitesdk.data.spi.predicates.Range;
+import org.kitesdk.data.spi.predicates.Ranges;
+
+@edu.umd.cs.findbugs.annotations.SuppressWarnings(value={
+    "NP_PARAMETER_MUST_BE_NONNULL_BUT_MARKED_AS_NULLABLE",
+    "SE_COMPARATOR_SHOULD_BE_SERIALIZABLE"},
+    justification="False positive due to generics.")
+@Immutable
+public class FixedLongRangeFieldPartitioner extends FieldPartitioner<Long, Long> {
+
+  private final long size;
+
+  public FixedLongRangeFieldPartitioner(String sourceName, long size) {
+    this(sourceName, null, size);
+  }
+
+  public FixedLongRangeFieldPartitioner(String sourceName, @Nullable String name,
+      long size) {
+    super(sourceName, (name == null ? sourceName + "_range" : name),
+        Long.class, Long.class);
+    this.size = size;
+    Preconditions.checkArgument(size > 0,
+        "Size of range buckets is not positive: %s", size);
+  }
+
+  @Override
+  public Long apply(Long value) {
+    return Math.round(Math.floor(value / ((double) size))) * size;
+  }
+
+  @Override
+  public Predicate<Long> project(Predicate<Long> predicate) {
+    if (predicate instanceof Exists) {
+      return Predicates.exists();
+    } else if (predicate instanceof In) {
+      return ((In<Long>) predicate).transform(this);
+    } else if (predicate instanceof Range) {
+      // must use a closed range:
+      //   if this( 5 ) => 10 then this( 6 ) => 10, so 10 must be included
```

Comment:
This projection method is correct, but there should be a test for it. Also, could you update or remove the comment on why it must be a closed range?